lib/ja_serializer/dsl.ex

```diff
@@ -0,0 +1,424 @@
+
+        def excerpt(post, _conn) do
+          [first | _ ] = String.split(post.body, ".")
+          first
+        end
+      end
+
+      post = %Post{
+        id: 1,
+        title: "jsonapi.org + Elixir = Awesome APIs",
+        body: "so. much. awesome.",
+        author: %Person{name: "Alan"}
+      }
+
+      post
+      |> PostSerializer.format
+      |> Poison.encode!
+
+  """
+
+  alias JaSerializer.Relationship.HasMany
+  alias JaSerializer.Relationship.HasOne
+
+  @doc false
+  defmacro __using__(_) do
+    quote do
+      @attributes []
+      @relations  []
+      @location   nil
+
+      import JaSerializer.DSL, only: [
+        attributes: 1, location: 1,
+        has_many: 2, has_one: 2, has_many: 1, has_one: 1
+      ]
+
+      unquote(define_default_attributes)
+      unquote(define_default_relationships)
+      unquote(define_default_links)
+
+      @before_compile JaSerializer.DSL
+    end
+  end
+
+  @doc false
+  defmacro __before_compile__(_env) do
+    quote do
+      def __relations,  do: @relations
+      def __location,   do: @location
+      def __attributes, do: @attributes
+    end
+  end
+
+  defp define_default_attributes do
+    quote do
+      def attributes(struct, conn) do
+        JaSerializer.DSL.default_attributes(__MODULE__, struct, conn)
+      end
+      defoverridable [attributes: 2]
+    end
+  end
+
+  @doc false
+  def default_attributes(serializer, struct, conn) do
+    serializer.__attributes
+    |> Enum.map(&({&1, apply(serializer, &1, [struct, conn])}))
+    |> Enum.into(%{})
+  end
+
+  defp define_default_relationships do
+    quote do
+      def relationships(struct, _conn) do
+        JaSerializer.DSL.default_relationships(__MODULE__)
+      end
+      defoverridable [relationships: 2]
+    end
+  end
+
+  @doc false
+  def default_relationships(serializer) do
+    serializer.__relations
+    |> Enum.map(&dsl_to_struct/1)
+    |> Enum.into(%{})
+  end
+
+  defp dsl_to_struct({:has_one, name, opts}),
+    do: {name, HasOne.from_dsl(name, opts)}
+  defp dsl_to_struct({:has_many, name, opts}),
+    do: {name, HasMany.from_dsl(name, opts)}
+
+  defp define_default_links do
+    quote do
+      def links(data, conn) do
+        JaSerializer.DSL.default_links(__MODULE__)
+      end
+      defoverridable [links: 2]
+    end
+  end
+
+  @doc false
+  def default_links(serializer) do
+    %{self: serializer.__location}
+  end
+
+  @doc """
+  Defines the canonical path for retrieving this resource.
+
+  ## String Examples
+
+  String may be either a relative or absolute path. Path segments beginning
+  with a colon are called as functions on the serializer with the struct and
+  conn passed in.
+
+      defmodule PostSerializer do
+        use JaSerializer
+
+        location "/posts/:id"
+      end
+
+      defmodule CommentSerializer do
+        use JaSerializer
+
+        location "http://api.example.com/posts/:post_id/comments/:id"
+
+        def post_id(comment, _conn), do: comment.post_id
```

Comment:
Is there a way to make this the default behavior? In this case it seems like you could just call comment.post_id if this function wasn't defined.